zenodo/modules/records/templates/zenodo_records/record_detail.html

```diff
@@ -27,19 +27,22 @@
 {%- from "zenodo_records/macros/accessright.html" import accessright_label -%}
 {%- from "zenodo_records/macros/objecttype.html" import objecttype_label -%}
 {%- from "zenodo_records/macros/altmetric.html" import almetric_badge -%}
-{%- from "zenodo_records/macros/authors.html" import authors -%}
+{%- from "zenodo_records/macros/authors.html" import authors as authors_macro -%}
 
 {%- set title = record.title + " | " + config.THEME_SITENAME -%}
 {%- set publication_date = record.publication_date|from_isodate -%}
 {%- set embargo_date = record.get('embargo_date', None)|from_isodate -%}
-{%- set object_type = record.resource_type|objecttype %}
+{%- set object_type = record.resource_type|objecttype -%}
+{%- set files = record.files if record.access_right == "open" else [] -%}
```

Comment:
If this is not working, then probably a template filter can do the job - e.g. ``{{record|accessible_files)}}``